torch/csrc/api/include/torch/nn/modules/linear.h

```diff
@@ -53,5 +53,51 @@ class TORCH_API LinearImpl : public Cloneable<LinearImpl> {
 /// module storage semantics.
 TORCH_MODULE(Linear);
 
+/// Options for the `Bilinear` module.
+struct TORCH_API BilinearOptions {
+  BilinearOptions(int64_t in1, int64_t in2, int64_t out);
+  /// The number of features in input 1 (columns of the input1 matrix).
+  TORCH_ARG(int64_t, in1);
+  /// The number of features in input 2 (columns of the input2 matrix).
+  TORCH_ARG(int64_t, in2);
+  /// The number of output features to produce (columns of the output matrix).
+  TORCH_ARG(int64_t, out);
+  /// Whether to learn and add a bias after the bilinear transformation.
+  TORCH_ARG(bool, with_bias) = true;
+};
+
+/// Applies a billinear transformation with optional bias.
+class TORCH_API BilinearImpl : public Cloneable<BilinearImpl> {
+ public:
+  BilinearImpl(int64_t in1, int64_t in2, int64_t out) : BilinearImpl(BilinearOptions(in1, in2, out)) {}
+  explicit BilinearImpl(BilinearOptions options);
```

Comment:
It would be awesome to change this to
```suggestion
  explicit BilinearImpl(const BilinearOptions& options_);
```
to match the design of other torch::nn modules.